src/pocketmine/entity/Entity.php

```diff
@@ -954,6 +958,15 @@ public function setHealth(float $amount) : void{
 
 		if($amount <= 0){
 			if($this->isAlive()){
+				if($this instanceof Player and $this->spawned and $this->hasTotemInHand()){
```

Comment:
`$this instanceof Player` makes declaring `Entity::consumeTotem()` useless. Maybe move this check into Human class? (See comment above)